src/crimson/osd/scrub/scrub_validator.h

```diff
@@ -0,0 +1,180 @@
+     [](const pg_stat_t &in) { return false; });
+  op("num_bytes"sv,
+     &object_stat_sum_t::num_bytes,
+     [](const pg_stat_t &in) { return false; });
+  op("num_object_clones"sv,
+     &object_stat_sum_t::num_object_clones,
+     [](const pg_stat_t &in) { return false; });
+  op("num_whiteouts"sv,
+     &object_stat_sum_t::num_whiteouts,
+     [](const pg_stat_t &in) { return false; });
+  op("num_objects_dirty"sv,
+     &object_stat_sum_t::num_objects_dirty,
+     [](const pg_stat_t &in) { return in.dirty_stats_invalid; });
+  op("num_objects_omap"sv,
+     &object_stat_sum_t::num_objects_omap,
+     [](const pg_stat_t &in) { return in.omap_stats_invalid; });
+  op("num_objects_pinned"sv,
+     &object_stat_sum_t::num_objects_pinned,
+     [](const pg_stat_t &in) { return in.pin_stats_invalid; });
+  op("num_objects_hit_set_archive"sv,
+     &object_stat_sum_t::num_objects_hit_set_archive,
+     [](const pg_stat_t &in) { return in.hitset_stats_invalid; });
+  op("num_bytes_hit_set_archive"sv,
+     &object_stat_sum_t::num_bytes_hit_set_archive,
+     [](const pg_stat_t &in) { return in.hitset_bytes_stats_invalid; });
+  op("num_objects_manifest"sv,
+     &object_stat_sum_t::num_objects_manifest,
+     [](const pg_stat_t &in) { return in.manifest_stats_invalid; });
+}
+
+/**
+ * iterate_scrub_maintained_stats
+ *
+ * For each scrub_maintained_stat member of object_stat_sum_t, invokes
+ * op with three arguments:
+ * - name of member (string_view)
+ * - pointer to member (T object_stat_sum_t::*)
+ * - skip for shallow (bool)
+ *
+ * Should be used to perform operations on all scrub_maintained_stat members
+ * such as updating the pg maintained instance once scrub is complete.
+ */
+template <typename Func>
+void foreach_scrub_maintained_stat(Func &&op) {
+  using namespace std::string_view_literals;
+  op("num_scrub_errors"sv, &object_stat_sum_t::num_scrub_errors, false);
+  op("num_shallow_scrub_errors"sv,
+     &object_stat_sum_t::num_shallow_scrub_errors,
+     false);
+  op("num_deep_scrub_errors"sv, &object_stat_sum_t::num_deep_scrub_errors, true);
+  op("num_omap_bytes"sv, &object_stat_sum_t::num_omap_bytes, true);
+  op("num_omap_keys"sv, &object_stat_sum_t::num_omap_keys, true);
+  op("num_large_omap_objects"sv,
+     &object_stat_sum_t::num_large_omap_objects,
+     true);
+}
+
+}
+
+template <>
+struct fmt::formatter<crimson::osd::scrub::chunk_result_t> {
+  constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
+
+  template <typename FormatContext>
+  auto format(
+    const crimson::osd::scrub::chunk_result_t &result, FormatContext& ctx) const
+  {
+    return fmt::format_to(
+      ctx.out(),
+      "chunk_result_t("
+      "num_scrub_errors: {}, "
+      "num_deep_scrub_errors: {}, "
+      "snapset_errors: [{}], "
+      "object_errors: [{}])",
+      result.stats.num_scrub_errors,
+      result.stats.num_deep_scrub_errors,
+      result.snapset_errors,
+      result.object_errors
+    );
+  }
+};
```

Comment:
Nice structure here. Unless I am missing something - it can be used in the classic code, too.

Reply:
Yeah, I figure that'll be the eventual goal.